deployment/src/main/java/org/wso2/testgrid/deployment/deployers/ShellDeployer.java

```diff
@@ -80,16 +83,32 @@ public DeploymentCreationResult deploy(TestPlan testPlan,
             if (exitCode > 0) {
                 logger.error(StringUtil.concatStrings("Error occurred while executing the deploy-provision script. ",
                         "Script exited with a status code of " , exitCode));
+                DeploymentCreationResult result = new DeploymentCreationResult();
+                result.setName(deploymentPatternConfig.getName());
                 result.setSuccess(false);
+                return result;
             }
         } catch (CommandExecutionException e) {
             throw new TestGridDeployerException(e);
         } catch (IOException e) {
             throw new TestGridDeployerException("Error occurred while retrieving the Testgrid_home ", e);
         }
-
+        DeploymentCreationResult result = DeploymentUtil.getDeploymentCreationResult(infrastructureProvisionResult
+                .getDeploymentScriptsDir());
         result.setName(deploymentPatternConfig.getName());
-        result.setHosts(infrastructureProvisionResult.getHosts());
+
+        List<Host> hosts = new ArrayList<>();
+        Host tomcatHost = new Host();
+        tomcatHost.setLabel("tomcatHost");
+        tomcatHost.setIp("ec2-52-54-230-106.compute-1.amazonaws.com");
+        Host tomcatPort = new Host();
+        tomcatPort.setLabel("tomcatPort");
+        tomcatPort.setIp("8080");
+        hosts.add(tomcatHost);
+        hosts.add(tomcatPort);
```

Comment:
Please create another issue for obtaining tomcat server details from either a configuration or database.

Reply:
Created #567